src/main/java/org/opensearch/repository/encrypted/EncryptedRepository.java

```diff
@@ -171,7 +175,8 @@ private EncryptionData createOrRestoreEncryptionData(final String clientName) th
                 );
             }
             encryptionData = encryptionDataGenerator.generate();
-            final byte[] bytes = encryptionDataSerializer.serialize(encryptionData);
+            final byte[] bytes =
+                    encryptionDataSerializer.serialize(encryptedRepositorySettings.encryptionProviderName(), encryptionData);
```

Comment:
The usage of the `encryptedRepositorySettings.encryptionProviderName()` is redundant here, the `encryptionDataSerializer` already accepts its in constructor:

```
        final EncryptionDataSerializer encryptionDataSerializer =
                new EncryptionDataSerializer(
                        encryptedRepositorySettings.encryptionProviderName(), <--- here
                        encryptedRepositorySettings.rsaKeyPair(clientName)
                );
```